File: frontend/src/types/escalation.ts
export interface EscalationStudent {
  id: string;
  nickname: string | null;
  email: string | null;
  faculty: string | null;
  year_of_study: number | null;
  is_anonymous: boolean;
}

export interface Escalation {
  id: string;
  user_id: string;
  message_id: string | null;
  level: string;
  status: 'open' | 'acknowledged' | 'resolved';
  distress_snapshot: Record<string, unknown> | null;
  counselor_notes: string | null;
  created_at: string;
  resolved_at: string | null;
  student: EscalationStudent | null;
}

File: backend/app/schemas/escalation.py
from typing import Literal

from pydantic import BaseModel, Field


class EscalationStudentSummary(BaseModel):
    id: str
    nickname: str | None
    email: str | None
    faculty: str | None
    year_of_study: int | None
    is_anonymous: bool


class EscalationResponse(BaseModel):
    id: str
    user_id: str
    message_id: str | None
    level: str
    status: str
    distress_snapshot: dict | None
    counselor_notes: str | None
    created_at: str
    resolved_at: str | None
    student: EscalationStudentSummary | None = None


class UpdateEscalationRequest(BaseModel):
    status: Literal["acknowledged", "resolved"] | None = None
    counselor_notes: str | None = Field(default=None, max_length=2000)
